src/components/GlobalPreferences/CustomLabels/LocalIdentities.js

```diff
@@ -52,204 +51,127 @@ function LocalIdentities({
   const compact = layoutName === 'small'
   const theme = useTheme()
 
+  if (!identities.length) {
+    return (
+      <Box>
+        <EmptyFilteredIdentities onClear={onClear} />
+      </Box>
+    )
+  }
+
   return (
-    <Box>
-      <div
+    <React.Fragment>
+      <Info
         css={`
-          display: grid;
-          grid-template-columns: auto auto auto;
-          grid-gap: ${1 * GU}px;
-          align-items: center;
-          justify-content: flex-end;
+          margin: 0 ${compact ? `${2 * GU}px` : 0} ${2 * GU}px;
           margin-bottom: ${2 * GU}px;
         `}
       >
-        <div
-          css={`
-            position: relative;
-          `}
-        >
-          <TextInput
-            adornment={
-              <IconSearch
-                css={`
-                  color: ${theme.surfaceOpened};
-                `}
-              />
-            }
-            adornmentPosition="end"
-            placeholder="Search"
-            onChange={onSearchChange}
-            value={searchTerm}
-            css={`
-              width: ${compact ? 25 * GU : 30 * GU}px;
-              ${textStyle('body2')};
-              color: ${searchTerm.trim() ? theme.surfaceContent : theme.hint};
-            `}
-          />
-        </div>
-        {!iOS && (
-          <Import
+        Any labels you add or import will only be shown on this device, and not
+        stored anywhere else. If you want to share the labels with other devices
+        or users, you will need to export them and share the .json file.
+      </Info>
+      <DataView
+        mode="table"
+        heading={
+          <Filters
+            searchTerm={searchTerm}
+            onSearchChange={onSearchChange}
             onImport={onImport}
-            button={
-              <Button
-                css={`
-                  ${compact &&
-                    `
-                      width: 50px;
-                      min-width: unset;
-                      padding: 0;
-                    `}
-                `}
-              >
-                <IconDownload
-                  css={`
-                    color: ${theme.surfaceOpened};
-                  `}
-                />
-                {!compact && (
-                  <span
-                    css={`
-                      display: inline-block;
-                      padding-left: ${1.5 * GU}px;
-                    `}
-                  >
-                    Import
-                  </span>
-                )}
-              </Button>
-            }
+            onShare={onShare}
+            onExport={onExport}
+            onRemove={onRemove}
+            someSelected={someSelected}
           />
+        }
+        selection={identities.reduce(
+          (p, { address }, index) => [
+            ...p,
+            ...(identitiesSelected.get(address) ? [index] : []),
```

Comment:
This needs an array of the selected indexes only, sometimes it may be a 1-1 where a map can be used but when these are not 1-1 a reduce is needed.
With your suggestion it is always a 1-1 no?

Reply:
Oh yes I understand now, I missed that `address` was nested in an object! Let’s keep the reduce :+1: